spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -294,58 +294,55 @@ public WriteApiResponse getWriteApiResponse(String tableName, InputStream jsonIn
     TableName parentTable =
         TableName.of(bigQuery.getOptions().getProjectId(), datasetName, tableName);
 
-    // Initialize a write stream for the specified table.
-    BigQueryJsonDataWriter writer = getBigQueryJsonDataWriter(parentTable);
-
-    try {
-      // Write data in batches. Ref: https://cloud.google.com/bigquery/quotas#write-api-limits
-      long offset = 0;
-      int currentBatchSize = 0;
-
-      BufferedReader jsonReader = new BufferedReader(new InputStreamReader(jsonInputStream));
-      String jsonLine = null;
-      JSONArray jsonBatch = new JSONArray();
-      while ((jsonLine = jsonReader.readLine()) != null) { // read the input stream line by line
-        JSONObject jsonObj = new JSONObject(jsonLine); // cast the JSON string into JSON Object
-        jsonBatch.put(jsonObj);
-        currentBatchSize++;
-        if (currentBatchSize
-            == getBatchSize()) { // append the batch, increment the offset and reset
-          // the batch
+    try (BigQueryJsonDataWriter writer = getBigQueryJsonDataWriter(parentTable)) {
+      try {
+        // Write data in batches. Ref: https://cloud.google.com/bigquery/quotas#write-api-limits
+        long offset = 0;
+        int currentBatchSize = 0;
+
+        BufferedReader jsonReader = new BufferedReader(new InputStreamReader(jsonInputStream));
+        String jsonLine = null;
+        JSONArray jsonBatch = new JSONArray();
+        while ((jsonLine = jsonReader.readLine()) != null) { // read the inputstream line by line
+          JSONObject jsonObj = new JSONObject(jsonLine); // cast the JSON string into JSON Object
+          jsonBatch.put(jsonObj);
+          currentBatchSize++;
+          if (currentBatchSize == getBatchSize()) {
+            // append the batch, increment the offset and reset the batch
+            writer.append(jsonBatch, offset);
+            offset += jsonBatch.length();
+            jsonBatch = new JSONArray();
+            currentBatchSize = 0;
+          }
+        }
+
+        if (jsonBatch.length() != 0) {
+          // there might be records less than JSON_STREAM_WRITER_BATCH_SIZE, append those as well
           writer.append(jsonBatch, offset);
-          offset += jsonBatch.length();
-          jsonBatch = new JSONArray();
-          currentBatchSize = 0;
         }
-      }
 
-      if (jsonBatch.length()
-          != 0) { // there might be records less than JSON_STREAM_WRITER_BATCH_SIZE, append those as
-        // well
-        writer.append(jsonBatch, offset);
+      } catch (Exception e) {
+        throw new BigQueryException("Failed to append records. \n" + e);
```

Comment:
Can we switch to using the `BigQueryException(String, Throwable)` constructor instead of converting the causing exception to a string?

Reply:
Done.